politeiawww/user/cockroachdb/models.go

```diff
@@ -48,6 +48,21 @@ func (User) TableName() string {
 	return tableUsers
 }
 
+// Session represents a politeiawww user session.
+type Session struct {
+	ID     uuid.UUID `gorm:"primary_key"`            // UUID
```

Comment:
That's right. We store a `UUID` in `string` form in the cookie (Gorilla session). In the domain model - if you will - we use a `UUID` for both a `User` and a `Session` identifier. That's why the code uses the latter.

Reply:
Where are we casting the session ID as a UUID?  Can you point to some code that I can look at?

Reply:
Sure

  1. [read `sessionid` from cookie](https://github.com/al-maisan/politeia/blob/0eebfded527a83895b5f6bc1a39d51efc499ca75/politeiawww/userwww.go#L60)
  2. [write `sessionid` to cookie](https://github.com/al-maisan/politeia/blob/0eebfded527a83895b5f6bc1a39d51efc499ca75/politeiawww/userwww.go#L168)

Reply:
> That's right. We store a `UUID` in `string` form in the cookie (Gorilla session). In the domain model - if you will - we use a `UUID` for both a `User` and a `Session` identifier. That's why the code uses the latter.

Let me clarify this a bit: a `uuid.UUID` instance is a stronger type than a `string` and using the former gives us a guarantee that we have a valid identifier which is not a given with a `string` since it may contain anything.

Reply:
IOW, we should probably change `getSessionUserID()` and `initSession()` to return / take a `uuid.UUID` as well.

Reply:
> IOW, we should probably change `getSessionUserID()` and `initSession()` to return / take a `uuid.UUID` as well.

scratch this -- almost everything uses `string` UUIDs

Reply:
It turns out we use `uuid.UUID` for primary / foreign keys in `politeiawww/user/cockroachdb/models.go` but the client code generally seems to be handling these UUID keys as strings.

Reply:
The [`Database` interface](https://github.com/al-maisan/politeia/blob/221be10130b2e05e752e25466f8b0be48552c8a6/politeiawww/user/user.go#L421) uses the `uuid.UUID` type for the `UserGetById()` function and I have copied the same style.

Reply:
> The Database interface uses the uuid.UUID type for the UserGetById() function and I have copied the same style.

Right, that's because the user ID is defined as a UUID everywhere.  Defining the session ID as a UUID when the gorilla package has it defined as a string adds mental overhead.  If there's is some technical reason, some benefit, that we're getting by defining it as a UUID then that may be fine, but if not, keep it the same as the gorilla package.

Reply:
Okay. Switched the `Session.ID` type to `string`.